### PR TITLE
refactor: store urns parts in search index builder efficiently

### PR DESCRIPTION
**Pull request by jywadhwani**
## Checklist
- [X] The PR conforms to DataHub's [Contributing Guideline](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [X] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable)

### CODE REVIEW

File: metadata-builders/src/main/java/com/linkedin/metadata/builders/search/DataProcessIndexBuilder.java [hunk 1]
```diff
@@ -62,23 +61,19 @@ private DataProcessDocument getDocumentToUpdateFromAspect(@Nonnull DataProcessUr
         return dataProcessDocument;
     }
 
-    @Nonnull
-    private DataProcessDocument getDocumentToUpdateFromAspect(@Nonnull DataProcessUrn urn, @Nonnull Status status) {
```
Comment (keremsahin1) [inline metadata-builders/src/main/java/com/linkedin/metadata/builders/search/DataProcessIndexBuilder.java:66]: Why is this removed?

File: metadata-builders/src/main/java/com/linkedin/metadata/builders/search/DatasetIndexBuilder.java [hunk 2]
```diff
@@ -51,26 +51,26 @@ private static DatasetDocument setUrnDerivedFields(@Nonnull DatasetUrn urn) {
...
 
   @Nonnull
   private DatasetDocument getDocumentToUpdateFromAspect(@Nonnull DatasetUrn urn, @Nonnull DatasetProperties datasetProperties) {
-    final DatasetDocument doc = setUrnDerivedFields(urn);
-    if (datasetProperties.hasDescription()) {
+    final DatasetDocument doc = new DatasetDocument();
+    if (datasetProperties.getDescription() != null) {
       doc.setDescription(datasetProperties.getDescription());
     } else {
       doc.setDescription("");
```
Comment (keremsahin1) [inline metadata-builders/src/main/java/com/linkedin/metadata/builders/search/DatasetIndexBuilder.java:76]: I also don't think we need this else here?

File: metadata-builders/src/main/java/com/linkedin/metadata/builders/search/DataProcessIndexBuilder.java (hunk 1 above)
Comment (jywadhwani) [reply metadata-builders/src/main/java/com/linkedin/metadata/builders/search/DataProcessIndexBuilder.java:66]: `Status` is not part of `DataProcessAspect` https://github.com/linkedin/datahub/blob/master/metadata-models/src/main/pegasus/com/linkedin/metadata/aspect/DataProcessAspect.pdl. This part of code is not being used.

File: metadata-builders/src/main/java/com/linkedin/metadata/builders/search/DatasetIndexBuilder.java (hunk 2 above)
Comment (jywadhwani) [reply metadata-builders/src/main/java/com/linkedin/metadata/builders/search/DatasetIndexBuilder.java:76]: Yes we don't, let me remove it.

**Review by keremsahin1 — APPROVED** [review]

**Comment by jplaisted** [comment]
FYI seems like this ended up _not_ setting the urn for most documents, causing the MAE processor to fail.

**Comment by jplaisted** [comment]
This also indicates our tests here are lacking.
